### ISSUE TITLE
[c] The last frame of an animation will mix back to the setup pose?

### ISSUE THREAD

**Issue by taigacon** [EsotericSoftware/spine-runtimes#814]
Relative to #812 , before take the advice, I tried to add a new animation in the editor, which just contains one key frame the same as the last frame of the animation I want to keep state with before.
Then I tried to use it in runtime with:
`setAnimation(0, "over", false);
addAnimation(0, "over2", true);`
However, the last frame of an animation "over" mixed back to the setup pose, and then it switched to the new animation "over2", everything seemed OK.

Here is the screenshot of the last frame of animation "over". I made a breakpoint at line 340 of AnimationState.c to be sure of breaking down on the last frame (before drawing).
![1](https://cloud.githubusercontent.com/assets/7021624/21764574/ce736046-d69e-11e6-9d28-f77e8af0532d.jpg)

Then I made a breakpoint at spAnimationState_update to meet the next frame, we can see the model had been updated and draw to the screen. It showed the setup pose.
![2](https://cloud.githubusercontent.com/assets/7021624/21764794/e7a31b82-d69f-11e6-84fc-2aa90b469a02.jpg)

And we meet the next frame again, the animation "over2" was playing and the first frame is OK.
![3](https://cloud.githubusercontent.com/assets/7021624/21764826/1926e9c2-d6a0-11e6-83e8-5d05e8579654.jpg)

Thank you for being tolerant of my poor English TAT. I have really tried my best.

**Comment by badlogic** [comment]
Could you provide me with your Spine project and the code you used to display the animation? It'd be best if you just modified a sample project in this repository so it's easier for me to follow what the problem is.

No worries about your Engoish, I'm not a native speaker either :)

**Comment by taigacon** [comment]
Here is the spine project, using Editor 3.5.46.

I don't know why github prevents me from uploading a zip file, just turn to google drive. ;)

https://drive.google.com/open?id=0B6F3biFZyXZaVkVUTzdjNURoUmM

![image](https://cloud.githubusercontent.com/assets/7021624/21769289/e905fc88-d6b8-11e6-9bb9-04d289565309.png)

**Comment by taigacon** [comment]
Oh, the code I used to display the animation has already written in the issue,
`setAnimation(0, "over", false); addAnimation(0, "over2", true);`

**Comment by badlogic** [comment]
Ok, I now see what the problem is. This is a known issue in the runtimes called the "dip problem". I've opened an issue for it here https://github.com/EsotericSoftware/spine-runtimes/issues/815. We have not found a good solution for it yet, however, in most scenarios there's a manual fix.

In your case, you can set the mix time between the animations "over" and "over2" to 0. When transitioning from "over" to "over2" the animation state will not mix to the setup pose. You can set the mix time for the two animations via `spAnimationStateData_setMixByName()` on the `spAnimationStateData` you passed to `spAnimatoinState_create()`.

Watch the other issue to get notified when we've found a fix for this issue.

**Comment by taigacon** [comment]
Thanks a lot!

**Comment by taigacon** [comment]
@badlogic I found that 

> set the mix time between the animations "over" and "over2" to 0

seems not working.

Here is the spine project:
https://drive.google.com/open?id=0B6F3biFZyXZaaVg2bHBZd3MwMFE

In the project there are two animations, 'animation' and 'state'.
Then I used
`setMix("animation","state",0)`
`setAnimation(0,"animation",false)`
`addAnimation(0, "state", true)`
but found that the dip problem is still there.

Here is the video:
https://drive.google.com/open?id=0B6F3biFZyXZaaGRfeHMtTjhrdGs
